consensus/enclave/impl/src/lib.rs

```diff
@@ -890,6 +892,13 @@ impl ConsensusEnclave for SgxConsensusEnclave {
                 &mint_tx.prefix.spend_public_key,
                 &mint_tx.prefix.view_public_key,
             );
+            if mint_tx.prefix.e_fog_hint.is_some()
```

Comment:
thank you, I agree

I'm going to leave the MOB check because I think Eran left it there as a duplicate check intentionally, to create an extra safeguard around accidentally minting MOB due to a bug.

but i'm going to move this new one to the validate_tx function as you suggest